HACKING.md

```diff
@@ -0,0 +1,473 @@
+    kubectl create deployment hello-server --image=$REGISTRY_IP:$REGISTRY_PORT/hello-app:1.0
+    kubectl get pod
+
+By repeating the last command we should after a few seconds see it turn to `Running` status for `hello-server-*`.
+If there is an error status, see [Troubleshooting](#troubleshooting) chapter.
+
+You can now remove the deployment:
+
+    kubectl delete deployment hello-server
+
+One important thing before deploying Strimzi Kafka Operator on Kind is to give the system account 'cluster-admin' permissions:
+
+    kubectl create clusterrolebinding strimzi-cluster-operator-cluster-admin --clusterrole=cluster-admin --serviceaccount=default:strimzi-cluster-operator
+
+
+### Starting and validating Strimzi Dev CLI
+
+In a new Terminal shell execute the following:
+
+```
+# Get internal configuration for access from within container:
+kind get kubeconfig --internal > ~/.kube/internal-kubeconfig
+
+# Make sure to use latest version of image
+docker pull quay.io/mstruk/strimzi-dev-cli
+
+# set DEV_DIR to directory where you have your cloned git repositories
+# You'll be able to access this directory from within Strimzi Dev CLI container
+export DEV_DIR=$HOME/devel
+
+# Now run the container
+docker run -ti --name strimzi-dev-cli -v /var/run/docker.sock:/var/run/docker.sock -v $HOME/.kube:/root/.kube -v $DEV_DIR:/root/devel -v $HOME/.m2:/root/.m2:cached quay.io/mstruk/strimzi-dev-cli /bin/sh
+```
+
+Note: If you exit the container or it gets shut down, as long as it's not manually deleted you can reattach and continue interactive session:
+
+    docker start strimzi-dev-cli
+    docker attach strimzi-dev-cli
+
+Having starterd the interactive session you are now in the development environment where you have all the necessary tools including `docker`, `kind`, `kubectl`, `git`, 'mvn' and all the rest you need to build Strimzi Kafka Operator components.
+
+Let's make sure that `docker`, and `kubectl` work:
+
+```
+export KUBECONFIG=~/.kube/internal-kubeconfig
+kubectl get ns
+docker ps
+```
+
+Also, let's make sure that we can push to local Docker Registry from Strimzi Dev CLI:
+
+```
+# Set REGISTRY_IP to the same value it has in the other Terminal session
+export REGISTRY_IP=<enter-the-ip-of-your-en0>
+
+export REGISTRY_PORT=5000
+
+# test docker push to local repository
+docker tag gcr.io/google-samples/hello-app:1.0 $REGISTRY_IP:$REGISTRY_PORT/hello-app:1.0
+docker push $REGISTRY_IP:$REGISTRY_PORT/hello-app:1.0
+
+```
+
+Building Strimzi Kafka OAuth
+----------------------------
+
+If you have not yet cloned the source repository it's time to do it now.
+It's best to fork the project and clone your fork, but you can also just clone the upstream repository.
+
+```
+cd /root/devel
+git clone https://github.com/strimzi/strimzi-kafka-oauth.git
+cd strimzi-kafka-oauth
+
+# Build it
+mvn clean spotbugs:check install
+
+# Make sure it's PR ready
+# Sorry, unfortunately the testsuite doesn't seem to be working inside strimzi-dev-cli
```

Comment:
Keeping it would signal to me and to users that we are aware of the issue and looking to address it. 
Otherwise people may try run it, and then ping us it doesn't work, to which we would be answering 'Yes we know'. Would be wasting their and our time.